src/common/lib/client/realtime.ts

```diff
@@ -161,6 +161,19 @@ class Channels extends EventEmitter {
     return channel;
   }
 
+  getDerived(name: string, deriveOptions: API.Types.DeriveOptions, channelOptions?: ChannelOptions) {
+    name = String(name);
+    if (deriveOptions.filter) {
+      const filter = encodeURIComponent(deriveOptions.filter)
+      const match = Utils.matchDerivedChannel(name)
+      name = `[filter=${filter}${match}`
+    }
+    if (channelOptions) {
+      return this.get(name, channelOptions)
+    }
+    return this.get(name)
```

Comment:
You can simplify this a bit - if `channelOptions` is undefined then `get()` will just ignore it either way
```suggestion
    return this.get(name, channelOptions)
```